pyiceberg/table/refs.py

```diff
@@ -34,6 +34,10 @@ def __repr__(self) -> str:
         """Return the string representation of the SnapshotRefType class."""
         return f"SnapshotRefType.{self.name}"
 
+    def __str__(self) -> str:
+        """Return the string representation of the SnapshotRefType class."""
+        return self.value
```

Comment:
Seems we need this `__str__`. Otherwise, 
```
>>> ref_type=SnapshotRefType.Branch
>>>f"Test: {ref_type}"
```
will show `Test: SnapshotRefType.Branch` in my local environment (Python 3.11) but `Test: branch` in CI's python 3.8